dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/requests/McpToolCallParams.java

```diff
@@ -71,32 +82,68 @@ public Map<String, Object> getArguments(Jsonb jsonb) {
         return parsedArguments;
     }
 
-    private Map<String, Object> parseArguments(JsonObject arguments, Jsonb jsonb) {
+    private Map<String, Object> parseArguments(JsonObject requestArguments, Jsonb jsonb) {
         Map<String, ArgumentMetadata> metadatas = metadata.arguments();
         Map<String, Object> result = new HashMap<>();
 
-        HashSet<String> argsProcessed = new HashSet<>();
-        for (var entry : arguments.entrySet()) {
-            String argName = entry.getKey();
-            JsonValue argValue = entry.getValue();
+        for (var argName : metadatas.keySet()) {
             ArgumentMetadata argMetadata = metadatas.get(argName);
-            if (argMetadata != null) {
-                String json = jsonb.toJson(argValue);
-                result.put(argName, jsonb.fromJson(json, argMetadata.type()));
+            if (requestArguments.containsKey(argName)) {
+                JsonValue argValue = requestArguments.get(argName);
```

Comment:
Don't call `containsKey` if you're immediately going to call `get`, unless the difference between the map containing a `null` value and the map not containing a value is important (in which case, please document that in a comment)
```suggestion
            JsonValue argValue = requestArguments.get(argName);
            if (argValue != null) {
```